extensions/wikia/EnableDiscussions/EnableDiscussionsHooks.class.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+
+class EnableDiscussionsHooks {
+
+	const DISCUSSIONS_URL = '/d/f';
+	// Forum namespaces should be available
+	// because both Forum.setup and ForumDisabled.setup define them
+	private static $forumNamespaces = [
+		NS_WIKIA_FORUM_BOARD,
+		NS_WIKIA_FORUM_TOPIC_BOARD,
+		NS_WIKIA_FORUM_BOARD_THREAD,
+	];
+
+
+	public static function onBeforeInitialize(
+		Title $title = null, $unused, OutputPage $output = null, $user, $request, $mediaWiki
+	) {
+		if ( !self::shouldRedirectToDiscussions() || $title == null ) {
+			return true;
+		}
+		$forumTitle = self::getForumTitle( $title );
+		if ( empty( $forumTitle ) ) {
+			return true;
+		}
+
+		$output->redirect( self::getDiscussionsUrl( $forumTitle->getNamespace(),
+			$forumTitle->getArticleID() ) );
+
+		return false;
+	}
+
+	/**
+	 * @return bool
+	 */
+	private static function shouldRedirectToDiscussions() {
+		global $wgEnableForumExt, $wgEnableDiscussions;
+
+		return empty( $wgEnableForumExt ) && $wgEnableDiscussions;
+	}
+
+	/**
+	 * @param $title
+	 * @return null|Title
+	 */
+	public static function getForumTitle( Title $title ) {
+		if ( $title->isSpecial( 'Forum' ) ||
+		     in_array( $title->getNamespace(), self::$forumNamespaces )
+		) {
+			return $title;
+		}
+		if ( $title->getNamespace() === NS_USER_WALL_MESSAGE ) {
+			$mainTitle = Title::newFromID( $title->getText() );
+			/*
+			 * if we visit the thread using Thread:xxxxx url, $wgTitle's namespace is 1201 for both, wall and forum
+			 * threads. But when we create new title from it using newFromId, we're able to distinguish those two
+			 */
+			if ( $mainTitle instanceof Title &&
+			     in_array( $mainTitle->getNamespace(), self::$forumNamespaces )
+			) {
+				return $mainTitle;
+			}
+		}
+		if ( $title->getNamespace() === 0 ) {
+			$mainTitle = Title::newFromID($title->getArticleID() );
+			if ($mainTitle instanceof Title &&
+			    in_array( $mainTitle->getNamespace(), self::$forumNamespaces )
+			) {
+				return $mainTitle;
+			}
+		}
+
+		return null;
+	}
+
+	private static function getDiscussionsUrl( $namespace, $id ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)